File: crates/re_types/src/components/text_log_level_ext.rs
use super::TextLogLevel;

impl TextLogLevel {
    /// Designates catastrophic failures.
    pub const CRITICAL: &str = "CRITICAL";

    /// Designates very serious errors.
    pub const ERROR: &str = "ERROR";

    /// Designates hazardous situations.
    pub const WARN: &str = "WARN";

    /// Designates useful information.
    pub const INFO: &str = "INFO";

    /// Designates lower priority information.
    pub const DEBUG: &str = "DEBUG";

    /// Designates very low priority, often extremely verbose, information.
    pub const TRACE: &str = "TRACE";

    #[inline]
    pub fn as_str(&self) -> &str {
        self.0.as_str()
    }
}

impl From<TextLogLevel> for String {
    #[inline]
    fn from(value: TextLogLevel) -> Self {
        value.as_str().to_owned()
    }
}

impl AsRef<str> for TextLogLevel {
    #[inline]
    fn as_ref(&self) -> &str {
        self.as_str()
    }
}

impl std::borrow::Borrow<str> for TextLogLevel {
    #[inline]
    fn borrow(&self) -> &str {
        self.as_str()
    }
}

File: rerun_py/rerun_sdk/rerun/components/text_log_level_ext.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from . import TextLogLevel


class TextLogLevelExt:
    """Extension for [TextLogLevel][rerun.components.TextLogLevel]."""

    CRITICAL: TextLogLevel = None  # type: ignore[assignment]
    """ Designates catastrophic failures. """

    ERROR: TextLogLevel = None  # type: ignore[assignment]
    """ Designates very serious errors. """

    WARN: TextLogLevel = None  # type: ignore[assignment]
    """ Designates hazardous situations. """

    INFO: TextLogLevel = None  # type: ignore[assignment]
    """ Designates useful information. """

    DEBUG: TextLogLevel = None  # type: ignore[assignment]
    """ Designates lower priority information. """

    TRACE: TextLogLevel = None  # type: ignore[assignment]
    """ Designates very low priority, often extremely verbose, information. """

    @staticmethod
    def deferred_patch_class(cls: Any) -> None:
        cls.CRITICAL = cls("CRITICAL")
        cls.ERROR = cls("ERROR")
        cls.WARN = cls("WARN")
        cls.INFO = cls("INFO")
        cls.DEBUG = cls("DEBUG")
        cls.TRACE = cls("TRACE")
